RecoJets/JetProducers/plugins/FixedGridRhoProducerFastjetFromRecHit.cc

```diff
@@ -0,0 +1,178 @@
+/*
+
+Author: Swagata Mukherjee
+
+Date: November 2021
+
+This producer computes rho from ECAL and HCAL recHits. 
+The current plan is to use it in egamma and muon HLT, who currently use 
+the other producer called FixedGridRhoProducerFastjet.
+At HLT, FixedGridRhoProducerFastjet takes calotowers as input and computes rho.
+But calotowers are expected to be phased out sometime in Run3.
+So this recHit-based rho producer, FixedGridRhoProducerFastjetFromRecHit, can be used as an alternative.
+
+*/
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloSubdetectorGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "CondFormats/EcalObjects/interface/EcalPFRecHitThresholds.h"
+#include "CondFormats/DataRecord/interface/EcalPFRecHitThresholdsRcd.h"
+#include "fastjet/tools/GridMedianBackgroundEstimator.hh"
+#include "RecoEgamma/EgammaIsolationAlgos/interface/EgammaHcalIsolation.h"
+
+class FixedGridRhoProducerFastjetFromRecHit : public edm::stream::EDProducer<> {
+public:
+  explicit FixedGridRhoProducerFastjetFromRecHit(const edm::ParameterSet &iConfig);
+  ~FixedGridRhoProducerFastjetFromRecHit() override;
+  static void fillDescriptions(edm::ConfigurationDescriptions &descriptions);
+
+private:
+  void produce(edm::Event &, const edm::EventSetup &) override;
+  void getHitP4(const DetId &detId, float hitE, math::XYZTLorentzVector &hitp4, const CaloGeometry &caloGeometry);
```

Comment:
```suggestion
  std::array<double, 4> getHitP4(const DetId &detId, float hitE, const CaloGeometry &caloGeometry) const;
```
(or float, in case everything is already float, I didn't check)